### PR TITLE
US132850 [AQ] Quiz Items Metrics apply settings. Export fix and refactoring

### PR DESCRIPTION
**Pull request by Vitalii-Misechko**
[US132850](https://rally1.rallydev.com/#/?detail=/userstory/611989499133&fdp=true): [AQ] Quiz Items Metrics apply settings

* Functional Testing 
  * [x] Unit tests
  * [x] Test cases
    * [x] Export to CSV for Quiz Details table works properly
    * [x] Export to CSV for Quiz Item table works properly
* Security, devops, perf: N/A
* UX and docs
  
FYI @Brightspace/draco-vis

### CODE REVIEW

**Comment by Vitalii-Misechko** [comment]
bump

**Review by NicholasHallman — APPROVED** [review]

**Review by bemailloux — APPROVED** [review]

File: pages/assessment-quality-dashboard-view.js [hunk 1]
```diff
@@ -480,25 +483,21 @@ class AssessmentQualityDashboardView extends SkeletonMixin(Localizer(MobxLitElem
...
+			{ valueFn: quiz => quiz.courseId },
+			{ valueFn: quiz => quiz.courseName },
+			{ valueFn: quiz => quiz.averageScore, formatFn: formatPercent },
+			{ valueFn: quiz => quiz.reliabilityCoefficient, formatFn: formatToFixed2 },
+			{ valueFn: quiz => quiz.numberOfQuizItems },
 			// TODO update when flaggedQuizItems implemented
-			// quiz.flaggedQuizItems
-			exportNullOrValue('numberOfLearnerAnswers'),
+			// { valueFn: quiz => quiz.flaggedQuizItems },
+			{ valueFn: quiz => quiz.numberOfLearnerAnswers }
```
[the remark this answers is not in the document]
Comment (Vitalii-Misechko) [reply pages/assessment-quality-dashboard-view.js:497]: Good point! I'll ask Kevin

File: model/exportData.js [hunk 2]
```diff
@@ -1,32 +1,31 @@
...
+		return headers.reduce((acc, val, i) => ({ ...acc, [val]: data[i] }), {});
+	});
+	exportFromJSON({ data: jsonArrayData, fileName: safeFileName, exportType: exportFromJSON.types.csv });
+}
 
-		return data[fieldName];
-	};
+export function formatRowForExport(data, exportFuncs, nullText) {
+	return exportFuncs.reduce((row, { valueFn, formatFn }) => {
+		if (!data || nullOrUndefinded(valueFn(data))) {
```
[the remark this answers is not in the document]
Comment (Vitalii-Misechko) [reply model/exportData.js]: > Multiple calls to `valueFn(data)` can be replaced with a variable

fixed

File: pages/assessment-quality-dashboard-view.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Vitalii-Misechko) [reply pages/assessment-quality-dashboard-view.js:497]: Kevin has decided to be consistent with ED. Therefore, we should export all columns regardless of settings

**Pull request merged**